lms/djangoapps/courseware/module_render.py

```diff
@@ -430,7 +439,7 @@ def handle_grade_event(block, event_type, event):  # pylint: disable=unused-argu
         grade = event.get('value')
         max_grade = event.get('max_value')
 
-        field_data_cache.set_score(
+        set_score(
```

Comment:
@cpennington @smarnach I know this is a bit outdated, but there's possible bug in this method now. 

So, since `set_score` does not assert on `user_id` being equal to current user id, it would be possible to send grade events with other users' ids. This might or might not be intentional, and in both cases there are problems:

1. If it is intentional, than  there's a problem with `_fulfill_content_milestones` invocation a couple of lines below, as it takes `user` from outer scope, rather than getting it by id. I'm not sure if that could cause trouble already, but still it *could* cause issues in future.
2. If it is not, than some validation/assertion should be added to avoid that, or, even better, `user_id` parameter should be removed from `grade` event.

Let me know what you think.

If you're interested in why I'm getting to this code: this commit is not there yet in solutions, and I've tried sending grade for other user. Resulted in [exception|https://github.com/open-craft/xblock-group-project-v2/pull/64#issuecomment-158072754]. I ended up removing the code that caused it, but it would be still good to know, if sending events for other users in is ok in general and for `grade` event in particular.